library/ziti_ctrl.c

```diff
@@ -237,9 +241,24 @@ static void ctrl_default_cb(void *s, const ziti_error *e, struct ctrl_resp *resp
     if (resp->new_address && strcmp(resp->new_address, ctrl->url) != 0) {
         CTRL_LOG(INFO, "controller supplied new address[%s]", resp->new_address);
 
+        const char *k;
+        ziti_controller_detail *detail;
+        MODEL_MAP_FOREACH(k, detail, &ctrl->endpoints) {
+            if (strcmp(k, ctrl->url) == 0) {
+                model_map_remove(&ctrl->endpoints, k);
+                break;
+            }
+        }
         FREE(ctrl->url);
         ctrl->url = resp->new_address;
         resp->new_address = NULL;
+        if(detail == NULL) {
+            detail = alloc_ziti_controller_detail();
+        }
+        FREE(detail->name);
```

Comment:
if detail was NULL previous, you still need to FREE  the name? seems like could be in the else?